(GH-1123) Allow users to configure apply_prep (#1157)
puppetlabs/bolt: pre-docs/bolt_configuration_options.md

`proxy`: The HTTP proxy to use for Git and Puppet Forge operations

`forge`: A subsection which can have its own `proxy` setting to set an HTTP proxy for only Puppet Forge operations, and a `baseurl` setting to specify a different Forge host

> ## Plugin hooks configuration options

Reviewer: maybe mention that `puppet_library` is the only configurable hook now though we'll likely to forget to keep that up to date.

I think most of the specifics about how this hook works should be in Install Libaries section.

`proxy`: The HTTP proxy to use for Git and Puppet Forge operations

`forge`: A subsection which can have its own `proxy` setting to set an HTTP proxy for only Puppet Forge operations, and a `baseurl` setting to specify a different Forge host

## Plugin hooks configuration options

> The `plugin_hooks` section allows you configure what plugins are used for a hook on a per target basis. This section is a hash where the keys are hooknames and the value specifies and configures the plugin to use. Currently there are two plugins you'll want to use for this: your own custom task using the `task` plugin, or the `install_agent` plugin which runs the `puppet_agent::install` task.

Reviewer 2: ```suggestion
The `plugin_hooks` section allows you to configure what plugins a specific hook should use for a specific target. This section is a hash where keys are hook names, and values specify and configure the plugin to use. Currently there are two possible plugins: the `task` plugin runs your own custom task, and the `install_agent` plugin runs the `puppet_agent::install` task.
```

> For now, the only configurable plugin hook is the `puppet_library` plugin hook.

Reviewer 2: ```suggestion
For now, the only configurable plugin hook is `puppet_library`.
```

```
plugin_hooks:
  puppet_library:
    plugin: install_agent
```

```

> plugin_hooks:

Reviewer: We usually prefer hyphens in the inventory file. That may be a bit confusing since the plugin names are `_`
